### PR TITLE
[serve] Fix flaky `test_http_state`

### PR DESCRIPTION
**Pull request by GeneDer**
## Why are these changes needed?

Ensure `state.update` is not called at terminal states.

## Related issue number

Patch https://buildkite.com/ray-project/oss-ci-build-branch/builds/5824#018a48cd-dead-40fc-82c0-547b3dab373f/6-322

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/serve/tests/test_http_state.py [hunk 1]
```diff
@@ -570,6 +570,10 @@ async def check_health(self):
     def _update_until_num_health_checks_received(
         state: HTTPProxyState, num_health_checks: int
     ):
+        # state.update should not be called in the terminal states.
+        if state.status in {ProxyStatus.UNHEALTHY, ProxyStatus.DRAINED}:
+            return True
```
Comment (edoakes) [inline python/ray/serve/tests/test_http_state.py:575]: is this guaranteed in the production codepath already? probably better to handle this case elegantly

Comment (GeneDer) [reply python/ray/serve/tests/test_http_state.py:575]: Or I guess we can just add an early return in the update so it becomes a no-op in those terminal states

Comment (edoakes) [reply python/ray/serve/tests/test_http_state.py:575]: that sounds better -- be defensive against ourselves

Comment (edoakes) [reply python/ray/serve/tests/test_http_state.py:575]: should also remove this early return then right?

**Review by edoakes — APPROVED** [review]

**Pull request merged**